src/coreclr/jit/codegenxarch.cpp

```diff
@@ -2554,6 +2554,148 @@ void CodeGen::genStackPointerDynamicAdjustmentWithProbe(regNumber regSpDelta)
     inst_Mov(TYP_I_IMPL, REG_SPBASE, regSpDelta, /* canSkip */ false);
 }
 
+//------------------------------------------------------------------------
+// genCodeForMemmove: Perform an unrolled memmove. The idea that we can
+//    ignore the fact that dst and src might overlap if we save the whole
+//    dst to temp regs in advance, e.g. for memmove(rax, rcx, 120):
+//
+//       vmovdqu  ymm0, ymmword ptr[rax +  0]
+//       vmovdqu  ymm1, ymmword ptr[rax + 32]
+//       vmovdqu  ymm2, ymmword ptr[rax + 64]
+//       vmovdqu  ymm3, ymmword ptr[rax + 88]
+//       vmovdqu  ymmword ptr[rcx +  0], ymm0
+//       vmovdqu  ymmword ptr[rcx + 32], ymm1
+//       vmovdqu  ymmword ptr[rcx + 64], ymm2
+//       vmovdqu  ymmword ptr[rcx + 88], ymm3
+//
+// Arguments:
+//    tree - GenTreeBlk node
+//
+void CodeGen::genCodeForMemmove(GenTreeBlk* tree)
+{
+    // Not yet finished for x86
+    assert(TARGET_POINTER_SIZE == 8);
```

Comment:
Just curious -- how do we expect the logic/heuristics to be different on x86?

Reply:
special byte-register for 1 byte access, up to 4 instead of 2 GPR for 1..15 range. I plan to file a follow up to add arm64 and then I'll probably add x86 and arm32 when I have time

Reply:
Why would we use more registers on x86, there are less registers available?

Reply:
> Why would we use more registers on x86, there are less registers available?

No, but in order to memmove 15 bytes you need 4*4b register to load it while on x64 you need 2*8b.
So I only implemented x64 to simplify code review for the initial version. Once it's merged it'd be eaiser to review a PR that adds x86